app/models/sms/decoder.rb

```diff
@@ -86,6 +86,21 @@ def find_form
     # if version not found, raise error
     raise_decoding_error("form_not_found", form_code: code) unless v
 
+    if @msg.mission
+      # if we already know the mission (it may or may not be already stored on the message)
+      # and it doesn't match the form's mission, complain
+      if v.form.mission != @msg.mission
+        raise_decoding_error("wrong_mission", form_code: code)
```

Comment:
Should we be giving away this information? Seems like saying that the form code is valid in another mission is a data leak. I'd prefer just to give whatever invalid message we would usually give if the form code does not exist in the mission.